code/modules/flufftext/Hallucination.dm

```diff
@@ -626,24 +727,44 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 		door_number = rand(0,4) //if 0 bolts all visible doors
 	var/count = 0
 	feedback_details += "Door amount: [door_number]"
+
 	for(var/obj/machinery/door/airlock/A in range(7, target))
 		if(count>door_number && door_number>0)
 			break
 		if(!A.density)
 			continue
 		count++
-		var/obj/effect/hallucination/fake_door_lock/lock = new(get_turf(A))
-		lock.target = target
-		lock.airlock = A
-		locks += lock
-		lock.lock()
-		sleep(rand(4,12))
-	sleep(100)
-	for(var/obj/effect/hallucination/fake_door_lock/lock in locks)
-		locks -= lock
-		lock.unlock()
-		sleep(rand(4,12))
-	qdel(src)
+		LAZYADD(airlocks_to_hit, A)
+
+	addtimer(VARSET_CALLBACK(src, unlocking, TRUE), 10 SECONDS)
+	START_PROCESSING(SSfastprocess, src)
+
+/datum/hallucination/bolts/process(delta_time)
+	next_action -= (delta_time * 10)
+	if (next_action > 0)
+		return
+
+	if (unlocking)
+		var/obj/effect/hallucination/fake_door_lock/next_unlock = popleft(locks)
+		if (next_unlock)
+			next_unlock.unlock()
+		else
```

Comment:
The order of locking -> unlocking is fucking with me, could you explain the logic behind it?

Reply:
I'm confused on what you mean, what bit are you asking for clarification on?

Reply:
why have the code to handle unlocking first rather then second, I wrote up a big screed asking about why you were iterating over an empty list, then I realized you were doing the effect creation in the second block, and the deletion in the first. IDK maybe I have brainworms

Reply:
I prefer to not have the first if statement be a negative, so for example:

```dm
if (!unlocking)
    // locking
else
    // unlocking
```

...would go against my personal code standards. In order to swap the order in that case, I'd need to inverse `unlocking` to be `locking`.

Reply:
Honestly it's real minor, I'd prefer it to be written in the same order as the work is done, but I accept that my monkey brain is flawed. Do what you'd like here